core/src/main/scala/org/bykn/bosatsu/Declaration.scala

```diff
@@ -645,25 +658,25 @@ object Declaration {
           }
 
         // one or more operators
-        val ops: P[Declaration => Operators.Formula[Declaration]] =
+        val ops: P[NonBinding => Operators.Formula[NonBinding]] =
           Operators.Formula.infixOps1(applied)
 
         // This already parses as many as it can, so we don't need repFn
         val form = ops.map { fn =>
 
-          { d: Declaration => convert(fn(d)) }
+          { d: NonBinding => convert(fn(d)) }
         }
 
-        maybeAp(applied, form)
+        NoCut(maybeAp(applied, form))
```

Comment:
this `NoCut` is due to the use of cuts inside operator parsing. This is due to ambiguity between `Foo(x) | Bar(y)` it could be an `operator |` or it could be a pattern. When we hit `Foo(x) | Bar(_, x)` we want to fail to parse that as a value because need to see that as one atomic pattern.